src/library/features/crates/cratefeaturehelper.cpp

```diff
@@ -67,9 +71,71 @@ CrateId CrateFeatureHelper::createEmptyCrate() {
     }
 
     CrateId newCrateId;
-    if (m_pTrackCollection->insertCrate(newCrate, &newCrateId)) {
+    if (m_pCrates->insertCrate(newCrate, &newCrateId)) {
         DEBUG_ASSERT(newCrateId.isValid());
         newCrate.setId(newCrateId);
+        m_pCrates->hierarchy().initClosureForCrate(newCrateId);
+        m_pCrates->hierarchy().generateCratePaths(newCrate);
```

Comment:
This logic should be moved to CrateManager. Again, the GUI code should know nothing about how the hierarchy is represented in the database.